Include/internal/pycore_object.h

```diff
@@ -54,16 +54,24 @@ PyAPI_FUNC(int) _PyObject_IsFreed(PyObject *);
    Furthermore, we can't use designated initializers in Extensions since these
    are not supported pre-C++20. Thus, keeping an internal copy here is the most
    backwards compatible solution */
+#if defined(Py_NOGIL)
+#define _PyObject_HEAD_INIT(type)                   \
+    {                                               \
+        .ob_ref_local = _Py_IMMORTAL_REFCNT_LOCAL,  \
+        .ob_type = (type)                           \
+    }
+#else
 #define _PyObject_HEAD_INIT(type)         \
     {                                     \
         .ob_refcnt = _Py_IMMORTAL_REFCNT, \
         .ob_type = (type)                 \
-    },
+    }
+#endif
 #define _PyVarObject_HEAD_INIT(type, size)    \
     {                                         \
-        .ob_base = _PyObject_HEAD_INIT(type)  \
+        .ob_base = _PyObject_HEAD_INIT(type), \
         .ob_size = size                       \
-    },
+    }
```

Comment:
Why I hate this comma, I dislike that you change it as part of a very more complicated change. Would you mind to keep the comma to reduce the size of the PR?

Reply:
I explained this in my first comment on this PR. It makes the PR smaller, not larger, because it avoids a number of `#ifdef Py_NOGIL` guards that would be necessary if the comma were still there. I put this as it's own commit (the first in the PR) so that the other changes can be reviewed separately.

Reply:
Oh ok, sorry, I didn't see your comment. Now it makes sense to me.